test/framework/machine_helpers.go

```diff
@@ -21,32 +21,66 @@ import (
 
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
-	"github.com/pkg/errors"
 
+	"github.com/pkg/errors"
 	clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha3"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+type GetMachinesByClusterInput struct {
+	Lister      Lister
+	ClusterName string
+	Namespace   string
+}
+
+// GetMachinesByCluster returns the Machine objects that matches provided options for a cluster.
+// Important! this method relies on labels that are created by the CAPI controllers during the first reconciliation, so
+// it is necessary to ensure this is already happened before calling it.
+func GetMachinesByCluster(ctx context.Context, input GetMachinesByClusterInput, options ...client.ListOption) []clusterv1.Machine {
```

Comment:
Either move options to `GetMachinesByClusterInput ` or create two methods:
- `GetMachinesByCluster`
- `GetMachinesByMachineDeployment`
I'm +1 for creating two methods

Reply:
+1 from me for separate methods as well.